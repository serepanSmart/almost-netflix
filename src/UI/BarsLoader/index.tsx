import React from 'react';
import styled, { keyframes } from 'styled-components';
import Colors from '../Theme/Colors';

export interface IBarsLoader {
  bgColor?: string;
  dataTestId?: string;
}

const Wrapper = styled.div`
  display: flex;
`;

const loading = keyframes`
  0% {
    transform: scale(1);
  }
  20% {
    transform: scale(1, 1.8);
  }
  40% {
    transform: scale(1);
  }
`;

export const Bar = styled.div<IBarsLoader>`
  display: inline-block;
  width: 4px;
  height: 18px;
  border-radius: 4px;
  margin: 0 2px;
  animation: ${loading} 1s ease-in-out infinite;
  &:nth-child(1) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0;
  }
  &:nth-child(2) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.09s;
  }
  &:nth-child(3) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.18s;
  }
  &:nth-child(4) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.27s;
  }
`;

const BarsLoader: React.FC<IBarsLoader> = ({
  bgColor = Colors.White,
  dataTestId = 'bars-loader'
}) => (
  <Wrapper data-testid={dataTestId}>
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
  </Wrapper>
);

export default BarsLoader;
