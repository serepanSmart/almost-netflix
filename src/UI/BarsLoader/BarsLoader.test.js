import { render } from '@testing-library/react';
import BarsLoader from '.';

const id = 'bars-loader';

describe('BarsLoader component', () => {
  let component;
  beforeEach(() => {
    component = (props) => render(<BarsLoader {...props} />);
  });

  test('should define BarsLoader component', () => {
    expect(component()).toBeDefined();
  });

  test('should be in the document', () => {
    const { getByTestId } = component();
    expect(getByTestId(id)).toBeInTheDocument();
  });

  test('should have attr "data-testid"', () => {
    const { getByTestId } = component({ dataTestId: id });
    expect(getByTestId(id)).toHaveAttribute('data-testid');
  });

  test('should match snapshot with custom color', () => {
    const bgColor = 'red';
    expect(component({ bgColor })).toMatchSnapshot();
  });
});
