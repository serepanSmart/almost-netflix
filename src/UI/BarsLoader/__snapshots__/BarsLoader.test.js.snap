// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarsLoader component should match snapshot with custom color 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: inline-block;
  width: 4px;
  height: 18px;
  border-radius: 4px;
  margin: 0 2px;
  -webkit-animation: ffhpKE 1s ease-in-out infinite;
  animation: ffhpKE 1s ease-in-out infinite;
}

.c1:nth-child(1) {
  background-color: red;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

.c1:nth-child(2) {
  background-color: red;
  -webkit-animation-delay: 0.09s;
  animation-delay: 0.09s;
}

.c1:nth-child(3) {
  background-color: red;
  -webkit-animation-delay: 0.18s;
  animation-delay: 0.18s;
}

.c1:nth-child(4) {
  background-color: red;
  -webkit-animation-delay: 0.27s;
  animation-delay: 0.27s;
}

<body>
    <div>
      <div
        class="c0"
        data-testid="bars-loader"
      >
        <div
          class="c1"
        />
        <div
          class="c1"
        />
        <div
          class="c1"
        />
        <div
          class="c1"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: inline-block;
  width: 4px;
  height: 18px;
  border-radius: 4px;
  margin: 0 2px;
  -webkit-animation: ffhpKE 1s ease-in-out infinite;
  animation: ffhpKE 1s ease-in-out infinite;
}

.c1:nth-child(1) {
  background-color: red;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

.c1:nth-child(2) {
  background-color: red;
  -webkit-animation-delay: 0.09s;
  animation-delay: 0.09s;
}

.c1:nth-child(3) {
  background-color: red;
  -webkit-animation-delay: 0.18s;
  animation-delay: 0.18s;
}

.c1:nth-child(4) {
  background-color: red;
  -webkit-animation-delay: 0.27s;
  animation-delay: 0.27s;
}

<div>
    <div
      class="c0"
      data-testid="bars-loader"
    >
      <div
        class="c1"
      />
      <div
        class="c1"
      />
      <div
        class="c1"
      />
      <div
        class="c1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
